packages/cms-lib/http.js

```diff
@@ -129,18 +147,25 @@ const createGetRequestStream = ({ contentType }) => async (
         },
         json: false,
       });
-      let response;
-      req
-        .on('error', reject)
-        .on('response', r => {
-          if (r.statusCode >= 200 && r.statusCode < 300) {
-            response = r;
-          } else {
-            reject(r);
-          }
-        })
-        .on('end', () => resolve(response))
-        .pipe(destination);
+      req.on('error', reject).on('response', res => {
+        if (res.statusCode >= 200 && res.statusCode < 300) {
+          const writeStream = fs.createWriteStream(filepath, {
+            encoding: 'binary',
+          });
+          req.pipe(writeStream);
+
+          writeStream.on('error', err => {
+            logFsError(err);
+            reject(err);
+          });
+          writeStream.on('close', async () => {
+            logger.log('Wrote file "%s"', filepath);
+            resolve(res);
+          });
+        } else {
+          reject(res);
+        }
+      });
```

Comment:
I wonder if a better solution might be to "clean up" whenever there is an error with either streaming the file to disk or the HTTP request (probably via `fs.unlink` or the synchronous equivalent). This would still involve creating the write stream but doing so before making the request.

Reply:
I did investigate that approach and it was a challenge with `--overwrite` because `createWriteStream` effectively overwrites the file immediately. Calling `unlink` does clean that up, but we've now removed an otherwise "good" file.

That would fix the issue cases where you're fetching a file for the first time, but doesn't help things if you're overwriting.

Reply:
Another approach would be to download to a `tmp` dir and move successfully fetched files over to the irdestination

Reply:
I think moving forward with the implementation in this PR is fine with me. I hadn't considered `--overwrite`. I am a bit curious about how others handle this but don't want to hold up a fix.

Reply:
Sounds good. I'm working on adding some tests around this and will get this in on monday